examples/http_server_proxy.rs

```diff
@@ -0,0 +1,65 @@
+// Run the example with:
+// cargo build --example http_server_proxy --target=wasm32-wasip2
+// wasmtime serve -Scli -Shttp --env TARGET_URL=https://example.com/ target/wasm32-wasip2/debug/examples/http_server_proxy.wasm
+// curl --no-buffer -v 127.0.0.1:8080/proxy/
+
+use wstd::http::body::Body;
+use wstd::http::{Client, Error, Request, Response, StatusCode, Uri};
+
+const PROXY_PREFIX: &str = "/proxy/";
+
+#[wstd::http_server]
+async fn main(server_req: Request<Body>) -> Result<Response<Body>, Error> {
+    match server_req.uri().path_and_query().unwrap().as_str() {
+        api_prefixed_path if api_prefixed_path.starts_with(PROXY_PREFIX) => {
+            // Remove PROXY_PREFIX
+            let target_url =
+                std::env::var("TARGET_URL").expect("missing environment variable TARGET_URL");
+            let target_url: Uri = format!(
+                "{target_url}{}",
+                api_prefixed_path
+                    .strip_prefix(PROXY_PREFIX)
+                    .expect("checked above")
+            )
+            .parse()
+            .expect("final target url should be parseable");
+            println!("Proxying to {target_url}");
+            proxy(server_req, target_url).await
+        }
+        _ => Ok(http_not_found(server_req)),
+    }
+}
+
+async fn proxy(server_req: Request<Body>, target_url: Uri) -> Result<Response<Body>, Error> {
+    let client = Client::new();
+    let mut client_req = Request::builder();
+    client_req = client_req.uri(target_url).method(server_req.method());
+
+    // Copy headers from `server_req` to the `client_req`.
+    for (key, value) in server_req.headers() {
+        client_req = client_req.header(key, value);
+    }
+
+    // Stream the request body.
+    let client_body = Body::from_http_body(server_req.into_body().into_boxed_body());
+    let client_req = client_req.body(client_body)?;
```

Comment:
Sorry, I dont know how I reviewed some out-of-date code here.

If you use `client_req.body(server_req.into_body())` here and skip the transformation to and from http_body::Body, wstd has an optimized path which will forward the body without copying the contents through the wasm guest memory. Same for the response body down below.